lib/runners/shell.rb

```diff
@@ -115,8 +118,13 @@ def capture3_trace(command, *args, **options)
         trace_writer.stderr stderr_str if trace_stderr
 
         if status.exited?
+          # normal case
           trace_writer.status status if trace_command_line
+        elsif status.termsig == SIGUSR2
+          # timeout
+          trace_writer.error "Analysis timeout (30 minutes)"
```

Comment:
[suggestion] Can we reuse `RUNNERS_TIMEOUT`?

https://github.com/sider/runners/blob/b28f6c63dc67310535a426a5b73a0cef26ab310a/images/docker-entrypoint.sh#L16

```suggestion
          trace_writer.error "Analysis timeout (#{ENV.fetch('RUNNERS_TIMEOUT')})"
```

Reply:
[question] Do we need `export RUNNERS_TIMEOUT` when the env var `RUNNERS_TIMEOUT` is unspecified?

```diff
--- a/images/docker-entrypoint.sh
+++ b/images/docker-entrypoint.sh
@@ -36,4 +36,6 @@ while [[ -n "$1" ]]; do
   shift
 done
 
+export RUNNERS_TIMEOUT
+
 exec /usr/bin/timeout --signal=SIGUSR2 --kill-after=10s "$RUNNERS_TIMEOUT" "${args[@]}"
```

Reply:
@gnakagaw ping?

Reply:
> Do we need...?

Yes. We need `export` declaration to pass `RUNNERS_TIMEOUT` to a child process (e.g. bin/runners).
I've implemented and pushed it.